cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -215,6 +217,17 @@ public ApplicationId start(@Name("namespaceId") NamespaceId namespace, AppReques
       throw new IllegalStateException("Preview service is not running. Cannot start preview for " + programId);
     }
 
+
+    Map<String, Object> config = appRequest.getConfig();
+    if (config.get(Constants.Preview.PREVIEW_REQUEST_NUM_OF_RECORDS) instanceof Double) {
```

Comment:
Why double? This configuration is integer. The best way it to check if it's `Number` and use `.intValue()`. This will handle any GSON representation for number